src/service/viewer-impl.js

```diff
@@ -872,8 +872,16 @@ export class Viewer {
       return Promise.resolve('');
     }
     return this.sendMessageUnreliable_('fragment', undefined, true).then(
-      hash => hash || ''
-    );
+        hash => {
+          if (!hash) {
+            return '';
+          }
+          /* Strip leading '#' */
+          if (hash.indexOf('#') == 0) {
```

Comment:
I can address this in the design doc whether to always or never come with a leading `#`. I think it's better to never come with a leading `#`. But for now I can validate it just in case.